x-pack/plugins/enterprise_search/server/lib/enterprise_search_http_agent.test.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { mockReadFileSync } from '../__mocks__/fs.mock';
```

Comment:
You actually don't need a separate file or mocks folder for this. You can simply do this inline at the top of the file like so:

```js
const mockReadFileSync = jest.fn();
jest.mock('fs', () => ({ readFileSync: mockReadFileSync }));
```

Alternatively, you can write it like this:

```js
jest.mock('fs', () => ({ readFileSync: jest.fn() }));
import { readFileSync } from 'fs';

// ...

expect(readFileSync).toHaveBeenCalledWith('some path', 'utf8');
```